### PR TITLE
starlight rewrite

### PR DESCRIPTION
**Pull request by DuroCodes**
rewrites the website from docusaurus -> starlight, please check thoroughly for any issues 🙏

### CODE REVIEW

**Comment by jacoobes** [comment]
remove .DS_Store

**Comment by jacoobes** [comment]
`git rm --cached src/**/*.DS_Store` i think, recommit and push

File: netlify.toml
Comment (jacoobes) [inline netlify.toml:1]: need this to deploy

File: .github/workflows/docusaurus.yml
Comment (jacoobes) [inline .github/workflows/docusaurus.yml:1]: need this as well to deploy. will need to switch it to deploy the astro starlight project

File: 404.html
Comment (jacoobes) [inline 404.html:1]: does astro have a default 404?

**Review by jacoobes — CHANGES_REQUESTED** [review]

File: .github/workflows/docusaurus.yml
Comment (jacoobes) [reply .github/workflows/docusaurus.yml:1]: i think its because the netlify preview not working so yea

Comment (jacoobes) [reply .github/workflows/docusaurus.yml:1]: idk 100% tho

File: 404.html
Comment (DuroCodes) [reply 404.html:1]: yeah:
image

File: .github/workflows/github-pages.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: Deploy to GitHub Pages
+
+on:
+  push:
```
Comment (SrIzan10) [inline .github/workflows/github-pages.yml:4]: friendly reminder for myself to add the `branches: [main]` rule when merging

**Comment by SrIzan10** [comment]
Are we ready to merge? I think I'm sending the @sernbot command

**Comment by jacoobes** [comment]
I will need to double check once more

Vào Th 3, 7 thg 5, 2024 lúc 9:11 SA sernbot ***@***.***> đã
viết:

**Comment by SrIzan10** [comment]
what the hell

**Comment by SrIzan10** [comment]
wait no fucking way

**Pull request merged**